mesonbuild/backend/ninjabackend.py

```diff
@@ -1103,6 +1103,22 @@ def generate_vala_compile(self, target, outfile):
                 # Install GIR to default location if requested by user
                 if len(target.install_dir) > 3 and target.install_dir[3] is True:
                     target.install_dir[3] = os.path.join(self.environment.get_datadir(), 'gir-1.0')
+            # Generate header for internal symbols if requested by user
+            if isinstance(target.vala_internal_header, str):
+                internalheadername = os.path.join(self.get_target_dir(target), target.vala_internal_header)
+                args += ['--internal-header', internalheadername]
+                valac_outputs.append(internalheadername)
+                target.outputs.append(target.vala_internal_header)
+                if len(target.install_dir) > 4 and target.install_dir[4] is True:
+                    target.install_dir[4] = self.environment.get_includedir()
```

Comment:
@powerwaremediacore if you use --hide-internal with valac these internal symbols will no longer be exported to the dynamic symbol table of the shared object. If you add this to your meson.build file I would expect the functionality you are seeing when using internal headers externally to no longer work. The autotools build of libgee already uses --hide-internal: https://git.gnome.org/browse/libgee/tree/configure.ac?h=meson#n23